server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java

```diff
@@ -119,6 +119,8 @@ public HttpClient get()
 
       httpClient.setIdleTimeout(config.getReadTimeout().getMillis());
       httpClient.setMaxConnectionsPerDestination(config.getNumConnections());
+      httpClient.setMaxRequestsQueuedPerDestination(config.getNumRequestsQueued());
+      httpClient.setConnectTimeout(500);
```

Comment:
hmmm... If you meant, AsyncQueryForwardingServlet.CANCELLATION_TIMEOUT_MILLIS then I wanted this to be independent really and not the same thing.
e.g. we could change this to be 600 ms or make configurable tomorrow if necessary but keep cancellation query timeout unchanged.